valhalla/midgard/sequence.h

```diff
@@ -44,8 +44,8 @@ class mem_map {
   mem_map(): ptr(nullptr), count(0), file_name("") { }
 
   //construct with file
-  mem_map(const std::string& file_name, size_t size): ptr(nullptr), count(0), file_name("") {
-    map(file_name, size);
+  mem_map(const std::string& file_name, size_t size, int advice = POSIX_FADV_NORMAL): ptr(nullptr), count(0), file_name("") {
```

Comment:
@dnesbitt61 @gknisely we should consider using `POSIX_FADV_SEQUENTIAL` these during graph creation and for the graphreader tar extract support. this provided a pretty decent performance increase when used to map small elevation tiles.

its mostly beneficial when you are reading a bunch of contiguous data from file, so for actual routing it might not be as good (all the edges of a node are next to each other but they may be small enough that the added read ahead doesnt matter). also when we are sorting the mapped file in OSM graph creation this will probably not be advantageous because we jump around randomly there. in that case we might try `POSIX_FADV_RANDOM`. anyway lots of fun stuff for someone with a rainy day of benchmarking in mind :)